### ISSUE TITLE
[Tensorflow GitHub] Do Inception-v3 Top1 Results match with the paper?

### ISSUE THREAD

**Issue by HossamAmer12** [tensorflow/tensorflow#25112]
This issue is to inquire about the top-1 rates of the inception-v3 results in the paper and available on the Tensorflow Github.

Currently, I am using the inception-v3 model [[link](http://download.tensorflow.org/models/image/imagenet/inception-2015-12-05.tgz) - inception-2015-12-05.tgz] from the official Tensorflow Github: [https://github.com/tensorflow/models/blob/master/tutorials/image/imagenet/classify_image.py](https://github.com/tensorflow/models/blob/master/tutorials/image/imagenet/classify_image.py)

When I calculate the top-1 accuracy rate for the 50K validation images in imageNet, I get 77.6%. However, I cannot find any reference to this result neither in the paper nor somewhere else.

Therefore, can you please help me get answers to the following question:
**What are the training specs of inception-2015-12-05.tgz? Single crop or basic or what? Are there references to its top1 rate results?** 

Please advise on my question as I believe this is a documentation issue that should be addressed. 

Please note that I am aware of this link:
[https://github.com/tensorflow/models/tree/master/research/slim]( https://github.com/tensorflow/models/tree/master/research/slim)
However, I am not also able to match the results from the updated link (inception_v3_2016_08_28.tar.gz - 78%) with the paper

I have seen a lot of links report different accuracies for the inception-v3 model - Please see some below:
[https://ai.googleblog.com/2016/08/improving-inception-and-image.html](https://ai.googleblog.com/2016/08/improving-inception-and-image.html)  [78% top1]
[https://github.com/tensorflow/models/tree/master/research/inception](https://github.com/tensorflow/models/tree/master/research/inception)  [78.8% top1]
[https://arxiv.org/pdf/1512.00567v1.pdf](https://arxiv.org/pdf/1512.00567v1.pdf) [76.6% top1]
https://medium.com/@sh.tsang/review-inception-v3-1st-runner-up-image-classification-in-ilsvrc-2015-17915421f77c [76.6% top1]
[](https://github.com/tensorflow/models/tree/master/research/slim#downloading-and-converting-to-tfrecord-format
) [78% top1]

**Comment by malzantot** [comment]
I  think what you are looking for is found in Table 2 of this paper: https://arxiv.org/abs/1512.00567

**Comment by HossamAmer12** [comment]
Thank you @malzantot for your response!

None of the results in the table are matching to 77.6% I produced on 50k imagNet validation set using [_inception-2015-12-05.tgz_] http://download.tensorflow.org/models/image/imagenet/inception-2015-12-05.tgz

Any clue? 
Also, do you know the training specs of _inception-2015-12-05.tgz_?

**Comment by HossamAmer12** [comment]
Any updates on this topic? 
@ymodak or anyone?

**Comment by ymodak** [comment]
@nealwu Can you PTAL? Thanks!

**Comment by nealwu** [comment]
Are you still facing this? If so perhaps @tfboyd would know?

**Comment by tfboyd** [comment]
This is not going to be a satisfying answer but it will be honest. I think you have two questions.  1) is there a model that exactly matches the paper and 2) your result with the tutorial does not match what is stated in the table, e.g. it ways 78% for 2016 and you are not getting that with the tutorial.

1) is there a model that matches the paper?
I do not know the inception v3 number very well. For say Resnet50, I have run it likely 1000 times and spent millions of dollars running it over and over and thus I know the variations and quirks.  The slim version is often the most accurate but their goal was not a perfect copy of the paper and they may exceed the accuracy of the paper due to whatever tweaks they may have made.  Unfortunately I do not think the SLIM authors do not publish the exact command used to train the model, e.g. what optimizer batch-size or whatever.

To draw a parallel to something I do know.  The ResNet50 paper has accuracy around 75% top_1 at maybe 120 epochs if my memory is correct, but no one runs that version they run what I have called V1.5 (which is not V2) and it trains to 76.0 to 76.5 from run to run at 90 epochs and nearly the same at 81. 

Having dealt with this in trying to setup performance tests it is almost impossible to find models that are identical to the paper or even identical to each other regardless of what they say the model is.   

This is a lot of words to say I doubt you will find out and the best way would be to try and get the attention of the people that trained the SLIM version. 

2) The tutorial is not giving you the accuracy from the table.
Your best bet is to open a ticket in models with the repro.  I have doubts it will get attention so I do not want to use your time.  This is a random and likely incorrect guess, but there are some images that a blacklisted for imagenet in the validation set and maybe they excluded those.  This is a long-shot, but an amusing thing to know.  I believe there are some images in imagenet labeled wrong in the validation set and some contests I think exclude them.  I do not think the difference is .4%, I think maybe .1% in the case of ResNet.  

I do not like not providing true answers but I think this is a good as you will get.  I hope this leads you toward something.  Closing due to I think this is the best you will get without some luck asking SLIM people in the tensorflow/models repo or someone digging into #2.
